minio/minio
# Fix healing delete marker

## Description
Healing was not working correctly in distributed mode because
errFileVersionNotFound was not properly converted in storage rest
client.

Besides, fixing healing delete marker is not working as expected.
## Motivation and Context
Fix healing in versioning mode


## How to test this PR?
1. Start a distributed setup, 4 nodes, 4 disks
2. Create a versioning bucket
3. Upload a file
4. Kill one node
5. Remove the file
6. Start the node again
Test 1:
`mc admin heal -r --scan=deep` command and `mc ls --versions` multiple times
Test 2:
Remove the object in disk 1, `mc admin heal -r scan=deep`, repeat with disk 2, 3 and 4 and check the integrity of the file.

## Types of changes

## Checklist:
